src/ConstantSum/ConstantSum.sol

```diff
@@ -133,28 +132,59 @@ contract ConstantSum is IStrategy {
     }
 
     // This should literally always work lol
-    function validateAllocateOrDeallocate(
+    function validateAllocate(
         address,
         uint256 poolId,
+        IDFMM.Pool calldata pool,
         bytes calldata data
     )
         external
         view
         returns (
             bool valid,
             int256 invariant,
-            uint256 reserveX,
-            uint256 reserveY,
-            uint256 totalLiquidity
+            uint256 deltaX,
+            uint256 deltaY,
+            uint256 deltaLiquidity
         )
     {
-        (reserveX, reserveY, totalLiquidity) =
+        (deltaX, deltaY, deltaLiquidity) =
```

Comment:
Oh I see. Only the allocations have the deltas. Perhaps we could also do this with swaps?

Reply:
Yes, it would be better to have all the functions expecting the same kind of parameters.